src/main/java/demo/util/EntityUtil.java

```diff
@@ -0,0 +1,80 @@
+package demo.util;
+
+import demo.annotation.Column;
+import demo.annotation.Entity;
+import demo.annotation.Id;
+import demo.annotation.Table;
+import demo.exception.MissingAnnotationException;
+import demo.session.EntityKey;
+import lombok.experimental.UtilityClass;
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+
+@UtilityClass
+public class EntityUtil {
+
+    public static EntityKey getEntityKey(Class<?> clazz, Object id) {
+        String entityName = clazz.getSimpleName();
+        String keyName;
+        Type keyType;
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field: declaredFields) {
+            if (field.isAnnotationPresent(Id.class)) {
+                keyName = field.getName();
+                keyType = field.getType();
+                return new EntityKey(entityName, keyName, id, keyType);
+            }
+        }
+        throw new MissingAnnotationException("The class '%s' doesn't have field annotated with @Id annotation"
+                .formatted(clazz.getSimpleName()));
+    }
+
+
+    public static Pair processEntity (Class<?> clazz) {
```

Comment:
@romanovosad87 the method is called `prosessEntity` and returns a `Pair`. It looks like a puzzle...
It should be called something like `createEntityMetadata`